pkg/tcl/testworkflowstcl/testworkflowprocessor/initprocess.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2024 Testkube.
+//
+// Licensed as a Testkube Pro file under the Testkube Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//	https://github.com/kubeshop/testkube/blob/main/licenses/TCL.txt
+
+package testworkflowprocessor
+
+import (
+	"errors"
+	"fmt"
+	"maps"
+	"strconv"
+	"strings"
+
+	testworkflowsv1 "github.com/kubeshop/testkube-operator/api/testworkflows/v1"
+	"github.com/kubeshop/testkube/internal/common"
+	"github.com/kubeshop/testkube/pkg/tcl/expressionstcl"
+)
+
+type initProcess struct {
+	ref        string
+	init       []string
+	params     []string
+	retry      map[string]testworkflowsv1.RetryPolicy
+	command    []string
+	args       []string
+	envs       []string
+	results    []string
+	conditions map[string][]string
+	negative   bool
+	errors     []error
+}
+
+func NewInitProcess() *initProcess {
+	return &initProcess{
+		conditions: map[string][]string{},
+		retry:      map[string]testworkflowsv1.RetryPolicy{},
+	}
+}
+
+func (p *initProcess) Error() error {
+	if len(p.errors) == 0 {
+		return nil
+	}
+	return errors.Join(p.errors...)
+}
+
+func (p *initProcess) SetRef(ref string) *initProcess {
+	p.ref = ref
+	return p
+}
+
+func (p *initProcess) Command() []string {
+	args := p.params
+
+	// TODO: Support nested retries
+	policy, ok := p.retry[p.ref]
+	if ok {
+		args = append(args, "--retryCount", strconv.Itoa(int(policy.Count)), "--retryUntil", expressionstcl.Escape(policy.Until))
```

Comment:
a list of constants shared with init image would help to make naming consistent between them

Reply:
`--retryCount` and `--retryUntil` are hackish for this moment, as they will be replaced in the final implementation. At the moment we support retry only on single-level, while the retry should be supported in multi-step structure as well:
```yaml
steps:
- name: Run something
  retry:
    count: 5
  steps:
  - name: Run test
    retry:
      count: 5 # totaling in max of 25 retries
  - name: Run another test
    retry:
      count: 5 # totaling in max of 25 retries minus the top retries used already by the previous step
```

We could considering adding some constants though, which would be imported by the `InitProcess` 👍

Reply:
Added constants and reused them (with `TODO` note for retries to delete it) 👍